src/transformStateWithClones.js

```diff
@@ -7,7 +7,57 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const NewArray = [];
+  let massive = { ...state };
+
+  for (let numArr = 0; numArr < actions.length; numArr++) {
+    const action = Object.assign({}, actions[numArr]);
+
+    for (const key in action) { // loop all types in array 'action'
+      if (action[key] === 'addProperties') {
+        const state1 = { ...massive };
```

Comment:
Why do you create this variable in each 'if'? You already have it at the beginning of the function body, no need to duplicate